lnclient/ldk/ldk.go

```diff
@@ -485,59 +485,68 @@ func (ls *LDKService) SendPaymentSync(ctx context.Context, invoice string, amoun
 	fee := uint64(0)
 	preimage := ""
 
-	for start := time.Now(); time.Since(start) < time.Second*50; {
-		event := <-ldkEventSubscription
-
-		eventPaymentSuccessful, isEventPaymentSuccessfulEvent := (*event).(ldk_node.EventPaymentSuccessful)
-		eventPaymentFailed, isEventPaymentFailedEvent := (*event).(ldk_node.EventPaymentFailed)
+	timeout := time.Second * time.Duration(timeoutSeconds)
```

Comment:
why implement it this way compared to just adding a simple case:
`case <-time.After(time * time.Second):`

Then we don't need a new timeout or the `if` do we inside `case <-ctx.Done():`?

It would be good to have 3 different cases:
- `ctx.Done` (the original context)
- `case <-time.After(time * time.Second):`
- `case ev := <-ldkEventSubscription:`

What do you think?

Reply:
Yeah but that would reset the timeout everytime you get an event. I changed it to use a timer instead, should be good now :+1: